pkg/asset/installconfig/installconfig.go

```diff
@@ -181,15 +181,15 @@ func (a *InstallConfig) finish(ctx context.Context, filename string) error {
 // that have already been checked by validation.ValidateInstallConfig().
 func (a *InstallConfig) platformValidation(ctx context.Context) error {
 	if a.Config.Platform.Azure != nil {
-		if a.Config.Platform.Azure.IsARO() {
-			// ARO performs platform validation in the Resource Provider before
-			// the Installer is called
-			return nil
-		}
 		client, err := a.Azure.Client()
 		if err != nil {
 			return err
 		}
+		if aro, err := client.CheckIfARO(context.TODO(), a.Config.Azure.ResourceGroupName); err == nil && aro {
+			// ARO performs platform validation in the Resource Provider before
+			// the Installer is called. Set isARO function value to true.
+			return nil
+		}
```

Comment:
Do we really need to skip validation for aro? Why? If it's validated once, it should be fine to validate again? Ideally the lofty goal would be to share common validation...